### ISSUE TITLE
ROS2 eloquent

### ISSUE THREAD

**Issue by Repp-cattt** [Project-MANAS/slam_gmapping#4]
Can I use it in the eloquent version of ros2? 
Or is there any other way for me to successfully build a map in ROS2?

**Comment by shiveshkhaitan** [comment]
Yes. Please check the [`eloquent-devel`](https://github.com/Project-Manas/slam_gmapping/tree/eloquent-devel) branch.

**Comment by Repp-cattt** [comment]
But when I use git clone to add eloquent-devel branch to download failed, in the source code did not find the version of eloquent, when I modified the source code compilation passed, but can not start the launch.py file normally, the error prompt that did not find the slam_gmapping package, this is why?


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 下午2:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
Yes. Please check the eloquent-devel branch.

**Comment by shiveshkhaitan** [comment]
I just added the `eloquent-devel` branch. Please try cloning the repo again.

**Comment by Repp-cattt** [comment]
My download order is&nbsp;https://github.com/Project-MANAS/slam_gmapping
My download order is git clone&nbsp;https://github.com/Project-MANAS/slam_gmapping --branch ros-eloquent-devel,I have also tried another method, that is&nbsp;git clone&nbsp;https://github.com/Project-MANAS/slam_gmapping/eloquent-devel&nbsp; ,They all failed.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 下午2:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
I just added the eloquent-devel branch. Please try cloning the repo again.

**Comment by shiveshkhaitan** [comment]
This command should work: `git clone https://github.com/Project-MANAS/slam_gmapping`. Please try again.

**Comment by Repp-cattt** [comment]
But your instruction did not specify a branch. I did use this instruction to download. When I finished compiling and wanted to run it, the system reported an error and did not find the slam_gmapping package.&nbsp;
In addition, I would like to ask a question, does this have anything to do with my inability to compile openslam_gmapping?



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 下午2:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
This command should work: git clone https://github.com/Project-MANAS/slam_gmapping. Please try again.

**Comment by shiveshkhaitan** [comment]
Yes I did not specify a branch because I have made `eloquent-devel` the default branch. You are probably getting error because that branch did not exist when you cloned the repo. That is why I am suggesting you to clone the repo once again. 

Yes `slam_gmapping` requires `openslam_gmapping` without which it would not build.

Please remember to source your local workspace `source install/setup.zsh` after building since `ros2 launch` does not automatically detect newly built packages.

**Comment by Repp-cattt** [comment]
I suddenly found that I could not use this command:&nbsp;
Git clone https://github.com/Project-MANAS/slam_gmapping.&nbsp;
To learn another way in Google, now use instructions:&nbsp;
Git clone git://github.com/Project-MANAS/slam_gmapping.&nbsp;
To download, does this have any effect?



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 下午2:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
This command should work: git clone https://github.com/Project-MANAS/slam_gmapping. Please try again.

**Comment by shiveshkhaitan** [comment]
That is ok. Should work as well

**Comment by Repp-cattt** [comment]
Thank you for your reply. I'll try again!&nbsp;
I wish you a happy life and smooth work ~ good night



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 下午3:11
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
That is ok. Should work as well

**Comment by Repp-cattt** [comment]
I'm sorry to bother you again, but after I ran the command "ros2 launch slam_gmapping slam_gmapping.launch.py", the expected map did not appear. Do I need to configure another radar topic / scan and odometer / odom topic?

**Comment by shiveshkhaitan** [comment]
Yes. As already mentioned in the `README.md`, it would require laser scans on topic `/scan` and the transforms (tf) between `odom` -> `base_link` -> `base_scan_link`.

**Comment by Repp-cattt** [comment]
I have changed the corresponding topic and transforms (tf), but when I run the "slam_gmapping.launch.py" file, I still can't see the map in rviz. Is it because I lack the robot_pose_ekf feature pack?&nbsp;
Later, I tried to use robot_pose_ekf in ros1 with gmapping through ros1_bridge, but I couldn't get the expected results.&nbsp;
I really can't figure out what went wrong.&nbsp;
I'm so sad._(:_」∠)_



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MANAS/slam_gmapping"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月17日(星期三) 下午5:54
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MANAS/slam_gmapping] ROS2 eloquent (#4)





 
Yes. As already mentioned in the README.md, it would require laser scans on topic /scan and the transforms (tf) between odom -&gt; base_link -&gt; base_scan_link.
